### PR TITLE
feat: add options to list rules

### LINKED ISSUE #37: List available rules and their description
Since it is not clear what rules are available it would be good to have an option to list rules and their descriptions, something like

```bash
timescale-doctor --list=* --show=brief
```

Option `--list` would take a glob pattern to match the rules to list, so `*` here means match all (and would be the default).

Option `--show` would select how much to show for each rule: just the description, brief or detailed messages, or even the query.

### PR DESCRIPTION
**Pull request by mkindahl**
The option `--list` is added that allow you to list rules matching a pattern and the option `--show` is present that allow you to pick the level of detail that you want to show for each rule.

Fixes #37

### CODE REVIEW

**Comment by mkindahl** [comment]
This is the output of `--help`

```
mats@fury:~/work/timescale/doctor$ timescale-doctor --help
usage: timescale-doctor [-U USER] [-d DBNAME] [-p PORT] [-h HOSTNAME] [--help] [--verbose] [--list [PATTERN]] [--show {brief,message,details}] [DBNAME]

Command-line interface to Timescale Doctor.

positional arguments:
  DBNAME                name of the database to connect to

options:
  -U USER, --username USER
                        user name to connect as
  -d DBNAME, --dbname DBNAME
                        name of the database to connect to
  -p PORT, --port PORT  database server port number
  -h HOSTNAME, --host HOSTNAME
                        database server host or socket directory
  --help                show this help message and exit
  --verbose, -v         Verbose logging. More options give higher verbosity.
  --list [PATTERN]      List rules matching pattern. If no pattern is given, will list all rules
  --show {brief,message,details}
```

**Comment by mkindahl** [comment]
This is the result of listing the rules:

```
mats@fury:~/work/timescale/doctor$ timescale-doctor --list
index.UnusedIndex:
    Find all unused indexes.
hypertable.HypertableCandidate:
    Detect candidate hypertable.
hypertable.ChunkPermissions:
    Detect bad chunk permissions.
compression.LinearSegmentBy:
    Detect segmentby column for compressed table.
compression.PointlessSegmentBy:
    Detect pointless segmentby column in compressed table.
```

**Comment by mkindahl** [comment]
This is how it looks when you select a section and show detailed and brief message.

```
mats@fury:~/work/timescale/doctor$ timescale-doctor --list=compression.* --show=details
compression.LinearSegmentBy:
    Detect segmentby column for compressed table.

    == MESSAGE ==

    Column '{attname}' in compressed hypertable '{relation}' has no
    distinct values.

    Column '{attname}' in hypertable '{relation}' as segment-by column
    is probably not a good choice since the number of values seems to
    grow with the number of rows of the table.

compression.PointlessSegmentBy:
    Detect pointless segmentby column in compressed table.

    == MESSAGE ==

    Column '{attname}' in hypertable '{relation}' is superfluous.

    Column '{attname}' in hypertable '{relation}' as segment-by column
    is pointless since it contains a single value.
```

File: src/doctor/cli.py [hunk 1]
```diff
@@ -42,10 +42,22 @@ def parse_arguments():
     parser.add_argument("--verbose", "-v", dest="log_level",
                         action="count", default=2,
                         help='Verbose logging. More options give higher verbosity.')
+    parser.add_argument("--list", nargs='?', const='*', metavar="PATTERN",
+                        help=("List rules matching pattern. "
+                              "If no pattern is given, will list all rules"))
+    parser.add_argument(
```
Comment (JamesGuthrie) [inline src/doctor/cli.py:48]: Looking at the code, and the usage, I didn't understand that `--show` should be used together with `--list`. I think that the program should abort if `--show` is used without `--list`.

**Review by JamesGuthrie — CHANGES_REQUESTED** [review]

File: src/doctor/cli.py (hunk 1 above)
Comment (mkindahl) [reply src/doctor/cli.py:49]: That is correct. Fixed that and here is the result:

```
mats@fury:~/work/timescale/doctor$ timescale-doctor --show=brief
usage: timescale-doctor [-U USER] [-d DBNAME] [-p PORT] [-h HOSTNAME] [--help] [--verbose] [--list [PATTERN]] [--show {brief,message,details}] [DBNAME]
timescale-doctor: error: called with '--show' but without '--list'
mats@fury:~/work/timescale/doctor$ timescale-doctor --list --show=brief
index.UnusedIndex:
    Find all unused indexes.
hypertable.HypertableCandidate:
    Detect candidate hypertable.
hypertable.ChunkPermissions:
    Detect bad chunk permissions.
compression.LinearSegmentBy:
    Detect segmentby column for compressed table.
compression.PointlessSegmentBy:
    Detect pointless segmentby column in compressed table.
mats@fury:~/work/timescale/doctor$ timescale-doctor --list=index.* --show=brief
index.UnusedIndex:
    Find all unused indexes.
mats@fury:~/work/timescale/doctor$ timescale-doctor --list=index.*
index.UnusedIndex:
    Find all unused indexes.
```

**Review by JamesGuthrie — APPROVED** [review]

**Pull request merged**